src/overview.md

```diff
@@ -0,0 +1,126 @@
+store your website's resources locally, on your users' devices. Consider Workbox
+if you want to:
+
+* Make your site work offline.
+* Improve load performance on repeat-visits. You can use Workbox to store
+  and serve common resources locally, rather than from the network.
+
+The only difference between the two options is whether you store all or only
+some resources locally.
+
+## The technology behind Workbox
+
+Modern offline web apps are possible thanks to **service workers**. Service
+workers are background workers, written in JavaScript, that can locally save
+all of the resources needed to run your site offline.
+
+You can think of a service worker as a [proxy server][proxy] that's running in
+the background of your users' devices. Your site makes a request for a resource,
+the service worker intercepts the request, and then decides whether to serve a
+local version of the resource or to fetch a fresh version from the network.
+
+[proxy]: https://en.wikipedia.org/wiki/Proxy_server
+
+The place where Workbox stores resources locally is called the **cache**.
+The act of storing locally is called **caching**. This is because the
+underlying technology that Workbox relies on is called the [Cache API][cache].
+This is completely different than [HTTP caching][http]. The main benefit
+that Workbox provides over HTTP caching is that it can cache resources *before*
+they're requested, whereas HTTP caching can only do so after the user goes to
+a page where the resource is required.
+
+[cache]: https://developer.mozilla.org/en-US/docs/Web/API/Cache
+[http]: https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching
+
+The logic for determining whether to serve a resource from the cache or to
+fetch a fresh version is called the **caching strategy** for that resource.
+It's common to use different caching strategies for different types of
+resources. For example:
+
+* For a logo that doesn't change often, you can probably serve it from the
+  cache.
+* For a JSON file containing a user's social media feed, you probably want
+  to display the cached content only while you fetch the latest content.
+
+## What problems Workbox solves
+
+In short, Workbox makes it easy to create optimal service worker code.
+
+Manually maintaining caches is tedious and full of pitfalls.
+The "core" Workbox libraries and tools --- `workbox-sw`, `workbox-build`,
+`workbox-webpack-plugin`, and `workbox-cli` --- make it easy to automatically
+generate a service worker that correctly caches resources.
+
+If you need to use different caching strategies for different types of
+resources, Workbox also simplifies this process. Much of the logic can be
+reduced to declarative build-time configurations. The rest of it can be
+abstracted away by calling the `workbox-sw` library from your custom
+service worker code.
+
+Workbox also intelligently manages updates to cached resources. In other words,
```

Comment:
There's a distinction here—Workbox will manage updated to **precached** resources. Resources that are cached at runtime (like API responses, or optional images) aren't managed in the same way. The update story for those resources depends on which runtime caching strategy is defined for their corresponding routes.

I think that's worth calling out, because there's a central distinction between the build-time precaching functionality vs. runtime caching functionality.